app/src/main/java/de/westnordost/streetcomplete/quests/postbox_collection_times/CollectionTimesAdapter.kt

```diff
@@ -64,13 +64,25 @@ class CollectionTimesAdapter(
     }
 
     fun addNew() {
+        addNewWeekdays()
+    }
+
+    fun addNewWeekdays() {
         val isFirst = collectionTimesRows.isEmpty()
         openSetWeekdaysDialog(getWeekdaysSuggestion(isFirst)) { weekdays ->
             openSetTimeDialog(12 * 60) { minutes ->
                 add(weekdays, minutes) }
         }
     }
 
+    fun addNewHours() {
+        val rowAbove = if (collectionTimesRows.size > 0) collectionTimesRows[collectionTimesRows.size - 1] else null
+        if (rowAbove !is WeekdaysTimesRow) return
```

Comment:
all rows of the `CollectinTimesAdapter` are `WeekdaysTimesRow`s, no need to check for that

Reply:
Resolved with e2a6116, yet I had to still check for `null` in place, otherwise Android Studio was complaining, that `rowAbove.weekdays` might not work.
https://github.com/westnordost/StreetComplete/pull/2109/files#diff-e6478a4cea0aeba97f040ea46df5f636R76
I think that's a reasonable precaution, right?